analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs

```diff
@@ -29,5 +31,75 @@ public class PublicMethodArgumentsShouldBeCheckedForNull : SymbolicRuleCheck
 
     protected override DiagnosticDescriptor Rule => S3900;
 
-    public override bool ShouldExecute() => false;
+    public override bool ShouldExecute()
+    {
+        return (IsRelevantMethod() || IsRelevantPropertyAccessor())
+            && IsAccessibleFromOtherAssemblies();
+
+        bool IsRelevantMethod() =>
+            Node is BaseMethodDeclarationSyntax { } method && MethodDereferencesArguments(method);
+
+        bool IsRelevantPropertyAccessor() =>
+            Node is AccessorDeclarationSyntax { } accessor
+            && (!accessor.Keyword.IsKind(SyntaxKind.GetKeyword) || accessor.Parent.Parent is IndexerDeclarationSyntax);
+
+        bool IsAccessibleFromOtherAssemblies() =>
+            SemanticModel.GetDeclaredSymbol(Node).GetEffectiveAccessibility() is
+                Public or Protected or ProtectedOrInternal;
```

Comment:
Fit's on single line, the break makes it harder to read
```suggestion
            SemanticModel.GetDeclaredSymbol(Node).GetEffectiveAccessibility() is Public or Protected or ProtectedOrInternal;
```